test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,224 @@
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specifications = $this->specificationLoader->list();
+
+        Assert::assertEquals($expectedSpecifications, $specifications);
+
+        $specificationConfig = $this->specificationLoader->get(self::SPECIFICATION_NAME);
+        Assert::assertEquals($expectedSpecificationConfig, $specificationConfig);
+    }
+
+    public function testLoadYamlFromCache(): void
+    {
+        $specificationFileName = 'specification.yaml';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specification = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+
+        rename(__DIR__ . '/' . $specificationFileName, __DIR__ . '/' . $specificationFileName . '_test');
+        $specificationFromCache = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+        Assert::assertSame($specification, $specificationFromCache);
+        rename(__DIR__ . '/' . $specificationFileName . '_test', __DIR__ . '/' . $specificationFileName);
+    }
+
+    public function testLoadJsonFromCache(): void
+    {
+        $specificationFileName = 'specification.json';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specification = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+
+        rename(__DIR__ . '/' . $specificationFileName, __DIR__ . '/' . $specificationFileName . '_test');
+        $specificationFromCache = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+        Assert::assertSame($specification, $specificationFromCache);
+        rename(__DIR__ . '/' . $specificationFileName . '_test', __DIR__ . '/' . $specificationFileName);
+    }
+
+    public function testLoadMissedSpecificationFileThrowsException(): void
+    {
+        $specificationFileName = 'specification.spec';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $this->expectException(Throwable::class);
+        $this->expectExceptionMessage(sprintf('The file "%s" does not exist.', __DIR__ . '/' . $specificationFileName));
+        $this->specificationLoader->load(self::SPECIFICATION_NAME);
+    }
+
+    public function testLoadForbiddenSpecificationTypeThrowsException(): void
+    {
+        $specificationFileName = 'specification.xml';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $this->expectException(Throwable::class);
```

Comment:
All possible exceptions that can be thrown should be tested, as they are all observable from outside of the class when interacting with it.